examples/gatekeeper/04-policy-constraint-templates.yml

```diff
@@ -0,0 +1,112 @@
+        plural: HTTPProxyRetryCountRanges
+        singular: HTTPProxyRetryCountRange
+      scope: Namespaced
+      validation:
+        openAPIV3Schema:
+          properties:
+            min:
+              type: integer
+            max: 
+              type: integer
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package httpproxy.retrycountrange
+
+        violation[{"msg": msg}] {
+          got := input.review.object.spec.routes[_].retryPolicy.count
+
+          got > input.parameters.max
+          msg := sprintf("retry policy count must be less than/equal to %v", [input.parameters.max])
+        }
+
+        violation[{"msg": msg}] {
+          got := input.review.object.spec.routes[_].retryPolicy.count
+
+          got < input.parameters.min
+          msg := sprintf("retry policy count must be greater than/equal to %v", [input.parameters.min])
+        }
+---
+# httpproxytimeoutrange is a ConstraintTemplate that enforces
+# min/max allowed timeout values for a specified field within
+# a route's timeoutPolicy on an HTTPProxy. Both min and max are
+# optional. 
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: httpproxytimeoutrange
+  labels:
+    app: contour
+spec:
+  crd:
+    spec:
+      names:
+        kind: HTTPProxyTimeoutRange
+        listKind: HTTPProxyTimeoutRangeList
+        plural: HTTPProxyTimeoutRanges
+        singular: HTTPProxyTimeoutRange
+      scope: Namespaced
+      validation:
+        openAPIV3Schema:
+          properties:
+            field:
+              type: string
+            min:
+              type: string
+            max: 
+              type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package httpproxy.timeoutrange
+
+        above_max = true {
+          # special-case for when field value is "infinity" -- max must be unspecified
+          input.review.object.spec.routes[_].timeoutPolicy[input.parameters.field] == "infinity"
+          input.parameters.max
+        } else = true {
+          # otherwise, compare durations 
+          got := input.review.object.spec.routes[_].timeoutPolicy[input.parameters.field]
+          time.parse_duration_ns(got) > time.parse_duration_ns(input.parameters.max)
+        }
+
+        below_min = false {
+          # special-case for when field value is "infinity" -- by definition is not below the min
+          input.review.object.spec.routes[_].timeoutPolicy[input.parameters.field] == "infinity" 
+        } else = true {
+          # otherwise, compare durations
+          got := input.review.object.spec.routes[_].timeoutPolicy[input.parameters.field]
+          time.parse_duration_ns(got) < time.parse_duration_ns(input.parameters.min)
+        }
+
+        violation[{"msg": msg}] {
+          above_max
+          msg := sprintf("timeout policy field '%s' must have value less than/equal to %v", [input.parameters.field, input.parameters.max])
```

Comment:
I think that this would make the validation more readable:

```
msg := sprintf("%q timeout must have value less than or equal to %q", [input.parameters.field, input.parameters.max])
```